Revert/RevertGui.java

```diff
@@ -0,0 +1,95 @@
+import java.util.*;
+import java.awt.event.ActionListener;
+import java.awt.event.ActionEvent;
+import javax.swing.JFrame;
+import javax.swing.JButton;
+import java.io.File;
+import java.io.FileReader;
+import java.io.BufferedReader;
+import java.io.IOException;
+import javax.swing.JLabel;
+import javax.swing.JTextField;
+import javax.swing.JOptionPane;
+
+
+public class RevertGui extends JFrame implements ActionListener 
+{
+	JButton bInvert, bShowLog;
+	JTextField tNum;
+	Config currentConfiguration;
+
+	
+	public RevertGui()
+	{
+		currentConfiguration = new Config();
+		currentConfiguration.readConfiguration();
+		
+		setLayout(null);
+		setSize(600,400);
+		setTitle("Revert");
+		
+		bInvert = new JButton("Invert the number!");
+		bInvert.setBounds(20,120,150,30);
+		bInvert.addActionListener(this);
+		add(bInvert);
+		
+		bShowLog = new JButton("Show logs");
+		bShowLog.setBounds(200,120,150,30);
+		bShowLog.addActionListener(this);
+		add(bShowLog);
+		
+		tNum = new JTextField("");
+		tNum.setBounds(20,40,150,30);
+		add(tNum);
+		
+	
+	}
+	
+	public void actionPerformed(ActionEvent e)

```

Comment:
Hmm, zamiast implementować tą klasą interfejs ActionListener i robić następnie addActionListener(this) na różnych elementach GUI, proponowałbym podejście, które nie będzie wymagać łańcucha `if` - `else if`:

1. Zdefiniuj metody, które będą obsługiwały poszczególne akcje, np. `private void onInvert(ActionEvent e)`
2. Dodaj `ActionListener`y w następujący sposób: `bInvert.addActionListener(this::onInvert)`

Reply:
Jak zrobiłem w ten sposób to mi nawet wyszło, że `bInvert` i `bShowLog` mogą być zadeklarowane jedynie w konstruktorze i wcale nie muszą być polami tej klasy :)